src/test/java/com/toopher/TestToopherAPI.java

```diff
@@ -1,89 +1,363 @@
+        assertEquals(terminalId, userTerminal.id);
+    }
+
+    @Test
+    public void testAdvancedUserTerminalsCreateWithExtras() throws InterruptedException, RequestError {
+        Map <String, String> extras = new HashMap<String, String>();
+        extras.put("foo", "bar");
+
+        HttpClientMock httpClient = new HttpClientMock(200, userTerminalJsonResponse.toString());
+        ToopherApi toopherApi = new ToopherApi("key", "secret",
+                createURI(DEFAULT_BASE_URL), httpClient);
+        UserTerminal userTerminal = toopherApi.advanced.userTerminals.create(userName, terminalName, terminalNameExtra, extras);
+
+        assertEquals("POST", httpClient.getLastCalledMethod());
+        assertEquals("user_terminals/create", httpClient.getLastCalledEndpoint());
+        assertEquals("bar", httpClient.getLastCalledData("foo"));
+        assertEquals(userName, httpClient.getLastCalledData("user_name"));
+        assertEquals(terminalName, httpClient.getLastCalledData("name"));
+        assertEquals(terminalNameExtra, httpClient.getLastCalledData("name_extra"));
+        assertEquals(terminalId, userTerminal.id);
+    }
+
+    @Test
+    public void testAdvancedUserTerminalsGetById() throws InterruptedException, RequestError {
+        HttpClientMock httpClient = new HttpClientMock(200, userTerminalJsonResponse.toString());
+        ToopherApi toopherApi = new ToopherApi("key", "secret",
+                createURI(DEFAULT_BASE_URL), httpClient);
+        UserTerminal userTerminal = toopherApi.advanced.userTerminals.getById(terminalId);
+
+        assertEquals("GET", httpClient.getLastCalledMethod());
+        assertEquals(String.format("user_terminals/%s", terminalId), httpClient.getLastCalledEndpoint());
+        assertEquals(terminalId, userTerminal.id);
+        assertEquals(userId, userTerminal.user.id);
+    }
+
+    @Test
+    public void testRawGet() throws InterruptedException, RequestError {
+        String endpoint = String.format("pairings/%s", id);
+
+        HttpClientMock httpClient = new HttpClientMock(200, pairingJsonResponse.toString());
+        ToopherApi toopherApi = new ToopherApi("key", "secret",
+                createURI(DEFAULT_BASE_URL), httpClient);
+        toopherApi.advanced.raw.get(endpoint);
+
+        assertEquals("GET", httpClient.getLastCalledMethod());
+        assertEquals(endpoint, httpClient.getLastCalledEndpoint());
+        assertEquals(pairingJsonResponse.toString(), httpClient.getExpectedResponse());
+    }
+
+    @Test
+    public void testRawPost() throws InterruptedException, RequestError {
+        String endpoint = "users/create";
+        List<NameValuePair> params = new ArrayList<NameValuePair>();
+        params.add(new BasicNameValuePair("name", userName));
+
+        HttpClientMock httpClient = new HttpClientMock(200, userJsonResponse.toString());
+        ToopherApi toopherApi = new ToopherApi("key", "secret",
+                createURI(DEFAULT_BASE_URL), httpClient);
```

Comment:
How do you feel about putting the `toopherApi` creation into the common test setup? (It seems to be repeated in every test.)